src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -322,17 +323,26 @@ public ConnectionDescription ConnectionDescription
                 CancellationToken cancellationToken)
             {
                 var readPreference = GetEffectiveReadPreference(slaveOk, null);
-                return Command(
+                var result = Command(
                     NoCoreSession.Instance,
                     readPreference,
                     databaseNamespace,
                     command,
+                    null, // commandPayloads
                     commandValidator,
                     null, // additionalOptions
-                    responseHandling,
+                    null, // postWriteAction
+                    CommandResponseHandling.Return,
                     resultSerializer,
                     messageEncoderSettings,
                     cancellationToken);
+
+                if (responseHandling != null && responseHandling() != CommandResponseHandling.Return)
+                {
+                    throw new NotSupportedException("This overload requires responseHandling to be: Return.");
```

Comment:
Alternatively we could require that the Func be null now and throw if it is not.